appGeo/src/app/snapshot/snapshot.component.ts

```diff
@@ -57,6 +58,46 @@ export class SnapshotComponent implements OnInit {
     this.snapshots = list;
   }
 
+  createSnapshot(){
+    console.log("Inside the create snapshot function")
+    let cm = new CampusMap();
+    databaseGet('game/map').then(value => {
+      console.log("val: " + JSON.stringify(value));
+    
+      if (value == null) {
+        const test_bub0 = new Bubble();
+        test_bub0.init_bubble("test_Bubble_without_map_in_db", 0, 0, 0, 0);
+        let testPlayer = new Player();
+        let loc = {longitude:20, latitude:20};
+        testPlayer.init(12, "player1", loc, ALIVE);
+        test_bub0.addPlayer(testPlayer);
+
+        var new_snap = new Snapshot(this.snapshots.length, test_bub0);
+        this.snapshots.push(new_snap);
+
+        console.log("new snapshot ID:", new_snap.snapshot_id)
+        console.log("new snapshot Bubble ID:", new_snap.snapshot_bubble_id)
+        console.log("new snapshot time:", new_snap.snapshot_time)
+        console.log("new snapshot content:", new_snap.snapshot_content)
+
+        databaseAdd('game/snapshots', this.snapshots);
+      } 
+      else {
+        cm = value;
+        var new_bub = cm.playersBubble(global.player);
+        var new_snap = new Snapshot(this.snapshots.length, new_bub);

```

Comment:
Just to clarify: This is the point where you are grabbing the map from the database --> getting the bubble information (which you then use to create the new snapshot)

Reply:
Yep, that's right. cm is the CampusMap object I grab from the database, and then playersBubble is a method inside the CampusMap class that takes in a player and returns the bubble that a player is in.

Reply:
This might help the killing and voting component because once you have the bubble that a player is in (call it new_bub), then you can go to new_bub.List to see the map of all players in that bubble.